I get this on a clean build:

```
Traceback (most recent call last):
  File "src/runServer.py", line 2, in <module>
    from flask.ext.sqlalchemy import SQLAlchemy
  File "/home/luke/cs4098/env/lib/python3.4/site-packages/flask/exthook.py", line 87, in load_module
    raise ImportError('No module named %s' % fullname)
ImportError: No module named flask.ext.sqlalchemy
make: *** [run] Error 1
```

--- a/makefile
+++ b/makefile
@@ -12,7 +12,7 @@ install: buildDeps
 	git submodule update
 	virtualenv -p /usr/bin/python3 env
 	env/bin/pip3 install Flask
-	env/bin/pip3 install SQLAlchemy
+	env/bin/pip3 install Flask-SQLAlchemy
 	env/bin/pip3 install Flask-OAuth
 	sudo apt-get install libncurses5-dev
 	cd peos/pml/ && make && \

--- a/src/runServer.py
+++ b/src/runServer.py
@@ -3,7 +3,7 @@ from flask.ext.sqlalchemy import SQLAlchemy
 # from flask.ext.login import LoginManager, UserMixin, login_user, logout_user,\
     # current_user
 # from flask_oauth import OAuth
-from oauth import OAuthSignIn
+# from oauth import OAuthSignIn
 from subprocess import check_output, STDOUT, CalledProcessError
 from werkzeug import generate_password_hash, check_password_hash, secure_filename
 

--- a/src/runServer.py
+++ b/src/runServer.py
@@ -1,8 +1,8 @@
 from flask import Flask, redirect, url_for, render_template,request, session, flash
 from flask.ext.sqlalchemy import SQLAlchemy
-from flask.ext.login import LoginManager, UserMixin, login_user, logout_user,\
-    current_user
-
+# from flask.ext.login import LoginManager, UserMixin, login_user, logout_user,\
+    # current_user
+# from flask_oauth import OAuth
 from oauth import OAuthSignIn
 from subprocess import check_output, STDOUT, CalledProcessError
 from werkzeug import generate_password_hash, check_password_hash, secure_filename
@@ -21,8 +21,21 @@ import parser
 
 DEBUG = True
 app = Flask(__name__)
+REDIRECT_URI = '/oauth'
 app.config['SECRET_KEY'] = 'secret'
-app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///db.sqlite'
+# app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///db.sqlite'
+# oauth = OAuth()
+# google = oauth.remote_app('google',
+#                           base_url='https://www.google.com/accounts/',
+#                           authorize_url='https://accounts.google.com/o/oauth2/auth',
+#                           request_token_url=None,
+#                           request_token_params={'scope': 'https://www.googleapis.com/auth/userinfo.email',
+#                                                 'response_type': 'code'},
+#                           access_token_url='https://accounts.google.com/o/oauth2/token',
+#                           access_token_method='POST',
+#                           access_token_params={'grant_type': 'authorization_code'},
+#                           consumer_key='305205879982-13behoi7hr2h9fnqd1t5sslgn43i972m.apps.googleusercontent.com',
+#                           consumer_secret='YdyrpHBahzLJ0qlPni6hy95d')
 app.config['OAUTH_CREDENTIALS'] = {
     'facebook': {
         'id': '604820106335654',
@@ -290,5 +303,61 @@ def oauth_callback(provider):
     if user is None:
         insert_social_user(social)
     return redirect('/')
+
+# @app.route('/google')
+# def index():
+#     access_token = session.get('access_token')
+#     if access_token is None:
+#         return redirect(url_for('googleLogin'))
+#
+#     access_token = access_token[0]
+#     headers = {'Authorization': 'OAuth '+access_token}
+#     req = Request('https://www.googleapis.com/oauth2/v1/userinfo', None, headers)
+#     try:
+#         res = urlopen(req)
+#     except URLError as e:
+#
+#         if e.code == 401:
+#             session.pop('access_token', None)
+#             return redirect(url_for('googleLogin'))
+#         return res.read()
+#     return res.read()
+
+# @app.route('/googleLogin')
+# def googleLogin():
+#     callback=url_for('_authorized', _external=True)
+#     return google.authorize(callback=callback)
+#
+# @app.route(REDIRECT_URI)
+# @google.authorized_handler
+# def _authorized(resp):
+#     access_token = resp['access_token']
+#     session['access_token'] = access_token, ''
+#     return redirect(url_for('index'))
+#
+# @google.tokengetter
+# def get_access_token():
+#     return session.get('access_token')
+
+
+def oauthRedirect(user_data,next_url):
+
+    email = user_data["email"]
+    print('EMAIL', email)
+    user = query_user(email)
+    session['email'] = email
+
+    if user == None:
+        insert_user(email)
+    else:
+        flash('Authentication failed.')
+        return redirect(url_for('login'))
+
+    return redirect(next_url)
+
+# @google.tokengetter
+# def get_access_token():
+#     return session.get('access_token')
+
 if __name__ == "__main__":
 	app.run(host="localhost", port=8000, debug=DEBUG)

--- a/src/runServer.py
+++ b/src/runServer.py
@@ -21,7 +21,7 @@ import parser
 
 DEBUG = True
 app = Flask(__name__)
-REDIRECT_URI = '/oauth'
+# REDIRECT_URI = '/oauth'
 app.config['SECRET_KEY'] = 'secret'
 # app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///db.sqlite'
 # oauth = OAuth()
@@ -340,20 +340,20 @@ def oauth_callback(provider):
 #     return session.get('access_token')
 
 
-def oauthRedirect(user_data,next_url):
-
-    email = user_data["email"]
-    print('EMAIL', email)
-    user = query_user(email)
-    session['email'] = email
-
-    if user == None:
-        insert_user(email)
-    else:
-        flash('Authentication failed.')
-        return redirect(url_for('login'))
-
-    return redirect(next_url)
+# def oauthRedirect(user_data,next_url):
+#
+#     email = user_data["email"]
+#     print('EMAIL', email)
+#     user = query_user(email)
+#     session['email'] = email
+#
+#     if user == None:
+#         insert_user(email)
+#     else:
+#         flash('Authentication failed.')
+#         return redirect(url_for('login'))
+#
+#     return redirect(next_url)
 
 # @google.tokengetter
 # def get_access_token():

--- a/src/templates/login.html
+++ b/src/templates/login.html
@@ -25,7 +25,7 @@
               <a href="{{ url_for('oauth_authorize', provider='facebook') }}">
                 <img src= "{{ url_for('static', filename = 'images/facebook.png') }}" width="50" height="50" alt="previous" title="Previous" border="0"/>
               </a>
-              <!-- <a href="#">
+              <!-- <a href="/googleLogin">
                 <img src= "{{ url_for('static', filename = 'images/google.png') }}" width="50" height="50" alt="previous" title="Previous" border="0"/>
               </a> -->
               <a href="{{ url_for('oauth_authorize', provider='twitter') }}">

